### PR TITLE
Use tox to support testing on Python 2.6 as well as 2.7.

### PR DESCRIPTION
**Pull request by Lukasa**
If we support Python 2.6 we should probably test on it.

Note that I haven't yet actually *run* this on Python 2.6, so the tests might fail! How exciting...

### CODE REVIEW

**Comment by Lukasa** [comment]
For those unfamiliar, `tox` is a tool that allows you to run your test suite in multiple different environments in one single command.

**Comment by Lukasa** [comment]
Hurrah, our tests do *not* pass on Python 2.6! This is because the following unittest methods don't exist:

- `assertRaisesRegexp`
- `assertIs`
- `assertIn`
- `assertItemsEqual`

Also, the argparse module does not appear to exist in the standard library on Python 2.6.

**Comment by Lukasa** [comment]
Backporting unittest2 and argparse gets the tests running, but now they fail because `deepcopy` doesn't like instance methods in Python 2.6, per [this known bug](http://bugs.python.org/issue1515).

**Comment by Lukasa** [comment]
The problem is that I don't know which object is defining a custom __new__. It doesn't seem to be any of ours though.

**Comment by Lukasa** [comment]
Alright, the actual problem is [this bug](http://bugs.python.org/issue2116): in Python 2.6, deepcopy of weakrefs isn't supported. There's a weakref deep in the fake table heirarchy, specifically from the chain to the table, so this causes us to bomb out when we try to deepcopy the table away in the iptables stub.

**Comment by Lukasa** [comment]
Next, in Python 2.6 we're vomiting DeprecationWarnings like no tomorrow:

```
/home/ubuntu/calico/calico/felix/felix.py:363: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
```

**Comment by Lukasa** [comment]
That turns out to be a legitimate problem that we have in Python 2.7 too, it's just that in Python 2.7 the `DeprecationWarning` is turned off by default.

**Comment by Lukasa** [comment]
retest this please

**Comment by Lukasa** [comment]
Victory is mine!

success

**Comment by Lukasa** [comment]
This is ready for review now, ideally from all of @neiljerram, @lwr20, and @plwhite.

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ tmp/
 .coverage
 /htmlcov
 /.dir-locals.el
+.tox/
```
Comment (neiljerram) [inline .gitignore]: Remove trailing slash, as not consistent with previous directory entries, e.g. htmlcov
Could also add a leading slash, to pin it to the top level directory, but I'm not really bothered.

File: MANIFEST.in [hunk 2]
```diff
@@ -0,0 +1 @@
+include README.md LICENSE openstack_plugin_requirements.txt acl_manager_requirements.txt felix_requirements.txt
```
Comment (neiljerram) [inline MANIFEST.in]: Add a newline?

File: calico/acl_manager/test/stub_acl_store.py [hunk 3]
```diff
@@ -12,7 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import unittest
+import sys
+
 from collections import defaultdict
```
Comment (neiljerram) [inline calico/acl_manager/test/stub_acl_store.py]: Not needed?

File: setup.py [hunk 4]
```diff
@@ -19,6 +19,9 @@
 import os
 import os.path
 import setuptools
+import sys
+
+PY2_DEPENDENCIES = ['argparse']
```
Comment (neiljerram) [inline setup.py]: Should that be PY26_DEPENDENCIES?

**Comment by neiljerram** [comment]
LGTM, modulo the notes I've made.  Please feel free to merge after looking at those.

**Comment by Lukasa** [comment]
All done. =)

**Comment by Lukasa** [comment]
And @neiljerram, I've updated run-unit-test to call into tox.

**Pull request merged**